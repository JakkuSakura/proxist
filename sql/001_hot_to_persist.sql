-- Setup: drop any existing tables to start clean
DROP TABLE IF EXISTS proxist.hot_ticks;
DROP TABLE IF EXISTS proxist.ticks_persist;

-- Create an in-memory table to simulate the hot engine
CREATE TABLE proxist.hot_ticks
(
    tenant String,
    symbol String,
    ts DateTime64(6),
    px Float64
) ENGINE = Memory;

-- Create a persistent table for cold storage
CREATE TABLE proxist.ticks_persist
(
    tenant String,
    symbol String,
    ts DateTime64(6),
    px Float64
) ENGINE = MergeTree
ORDER BY (tenant, symbol, ts);

-- Seed hot data
INSERT INTO proxist.hot_ticks VALUES
('alpha', 'AAPL', toDateTime64('2024-01-01 12:00:00', 6), 100.75),
('alpha', 'AAPL', toDateTime64('2024-01-01 12:00:01', 6), 100.50),
('alpha', 'MSFT', toDateTime64('2024-01-01 12:00:02', 6), 99.50);

-- Persist the in-memory rows into the durable table
INSERT INTO proxist.ticks_persist
SELECT tenant, symbol, ts, px
FROM proxist.hot_ticks;

-- Report results for verification
SELECT tenant, symbol, count() AS rows, round(sum(px), 2) AS total_px
FROM proxist.ticks_persist
GROUP BY tenant, symbol
ORDER BY tenant, symbol
FORMAT TSVWithNames;

