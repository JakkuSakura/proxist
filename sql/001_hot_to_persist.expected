tenant	symbol	rows	total_px
alpha	AAPL	2	201.25
alpha	MSFT	1	99.5
